src/dataset-builder/ConceptSearch.ts

```diff
@@ -34,15 +35,23 @@ const DebouncedTextInput = withDebouncedChange(TextInput);
 export const ConceptSearch = (props: ConceptSearchProps) => {
   const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
     props;
-  const [search, setSearch] = useState<string>(initialSearch);
+  const [searchText, setSearchText] = useState<string>(initialSearch);
   const [cart, setCart] = useState<Concept[]>(initialCart);
   const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
 
-  useEffect(() => {
+  // when mounting ConceptSearch, we want to retrieve a list of searchConcepts
+  useOnMount(() => {
     void searchConcepts(() => {
-      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root);
```

Comment:
I think this is incorrect, in the case where `initialSearch` contains a search string. This can occur when the user navigates back to the search page from the hierarchy page, and they had a search string entered before.